### PR TITLE
RHODS-8272: Setup repo for Openshift CI

### PR DESCRIPTION
**Pull request by RobGeada**
Adds:

- Build+Push github action to Quay for PR images, tagged as trustyai-service-ci:$SHA
- OpenShift-CI test repo, to be integrated into Openshift-CI system 
- Restructures TrustyAI manifests to allow the image name to be specified from the kfdef, rather than in a file in a committed branch of a github repo.
- Adds a `rhods` overlay to the TrustyAI deployment, to automatically select the latest nightly RHODS image

Many thanks for submitting your Pull Request :heart:!

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/trustyai-explainability/trustyai-explainability/blob/main/CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `FAI-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.3.x] FAI-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: .github/workflows/build-and-push.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+          docker pull $(cat Dockerfile | grep -o -P '(?<=FROM ).*(?= AS runtime)')
+          docker build -t ${{ env.IMAGE_NAME }}:ci-$SHA .
+      - name: Log in to Quay
+        run: docker login -u ${{ secrets.QUAY_ROBOT_USERNAME }} -p ${{ secrets.QUAY_ROBOT_SECRET }} quay.io
+      - name: Push to Quay
+        run: |
+          # Strip git ref prefix from version
+          # Currently, we rely on nightly builds for pushing images, so exit if
+          # not a tag version for now.
+          if [ "$BRANCH_NAME" == "main" ]; then
```
Comment (danielezonca) [inline .github/workflows/build-and-push.yaml]: Is there a reason why this filter is done only after the creation of the image?
If we want to skip all branch except for `main` it is possible to do it directly in the `on` section
```
on:
  push:
    branches:
      - main
    paths-ignore:
      - 'LICENSE'
      - '**/.gitignore'
      - '**.md'
      - '**.adoc'
      - '*.txt'
      - '.github/**'
  pull_request_target:
    paths-ignore:
      - 'LICENSE'
      - '**/.gitignore'
      - '**.md'
      - '**.adoc'
      - '*.txt'
      - '.github/**'
```

File: explainability-service/manifests/overlays/rhods/params.env [hunk 2]
```diff
@@ -0,0 +1 @@
+image=registry-proxy.engineering.redhat.com/rh-osbs/managed-open-data-hub-odh-trustyai-service-rhel8:latest
```
Comment (danielezonca) [inline explainability-service/manifests/overlays/rhods/params.env]: Is it possible to plug this value from outside? This is an internal registry

File: explainability-service/manifests/base/trustyai-deployment.yaml [hunk 3]
```diff
@@ -10,6 +10,7 @@ metadata:
     prometheus.io/path: /q/metrics
     prometheus.io/port: "8080"
     prometheus.io/scheme: http
+    service.alpha.openshift.io/serving-cert-secret-name: trustyai-service-proxy-tls
```
Comment (danielezonca) [inline explainability-service/manifests/base/trustyai-deployment.yaml]: Is this related with the PR?
It looks related to TLS configuration but I don't find other references

File: tests/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,68 @@
+FROM quay.io/centos/centos:stream8
```
Comment (danielezonca) [inline tests/Dockerfile:1]: I find strange that we use `centos` as base image but if this is copied by ODH I'm fine to align :)

File: tests/resources/trustyai/model.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+  name: example-sklearn-isvc
+  annotations:
+    serving.kserve.io/deploymentMode: ModelMesh
+spec:
+  predictor:
+    model:
+      modelFormat:
+        name: sklearn
+      runtime: mlserver-0.x
+      storageUri: "https://github.com/trustyai-explainability/trustyai-explainability/raw/main/explainability-service/demo/models/model.joblib?raw=true"
```
Comment (danielezonca) [inline tests/resources/trustyai/model.yaml]: It might be useful in the future to make this "stable" to a fixed commit but I don't expect this model to change often so let's try in this way and see how it works

**Review by danielezonca — COMMENTED** [review]
Overall looks good to me, just some comments

File: .github/workflows/build-and-push.yaml (hunk 1 above)
Comment (RobGeada) [reply .github/workflows/build-and-push.yaml]: Addressed in slack

**Review by danielezonca — APPROVED** [review]
Thanks for the PR, LGTM :+1:
